samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -141,7 +142,7 @@ class OffsetManager(
    * SystemAdmins that are used to get next offsets from last checkpointed
    * offsets. Map is from system name to SystemAdmin class for the system.
    */
-  val systemAdmins: Map[String, SystemAdmin] = Map(),
+  val systemAdmins: SystemAdmins = new SystemAdmins(),
```

Comment:
Does this work correctly? Looks like OffsetManager#apply is explicitly passing a null SystemAdmins if it's not passed one. Maybe change the default there to a new SystemAdmins() instead of null.

Reply:
Good point. I have updated the `OffsetManager#apply` as suggested.